### ISSUE TITLE
export $(dotenvx get --format=shell) doesn't handle values with spaces well

### ISSUE THREAD

**Issue by bijeebuss2** [dotenvx/dotenvx#405]
If I have .env
```
A="one two three"
```
and I run 
```
export $(dotenvx get --format=shell)
echo $A
```
I get `one` and I expect `one two three`

**Comment by motdotla** [comment]
yeah I would too. i'm starting to regret adding these shell formatters because shells are very tricky here.

here's what is happening:

```sh
$ dotenvx get --format=shell
A=one two three
```

that equates to this:

```sh
$ A=one two three echo $A
bash: two: command not found
```

obviously, I'd never write it like that and expect it to work, instead I'd write it like the following:

```sh
$ A="one two three" echo $A
one two three
```

so why doesn't `$ dotenvx get --format=shell` return with quotes? because then it breaks `env` by including the quotes! it's very frustrating! first how it works currently with `env`:

```
$ env $(dotenvx get --format=shell)
...
A=one two three
```

^ this correctly prints `one two three`

but if quotes are returned in the subshell command, `env` preserves the quotes! here's the equivalent example:

```
$ env 'A="one two three"'
...
A="one two three"
```

the only truly reliable solution right now is to use `eval` with `--format=eval`.

```
eval $(dotenvx get --format=eval)
```

i'm open to ideas and conversation around this though. it's probably the most frustrating part of this library.

**Comment by motdotla** [comment]
related

* https://github.com/dotenvx/dotenvx/pull/393
* https://github.com/dotenvx/dotenvx/issues/392
* https://github.com/dotenvx/dotenvx/pull/366

**Comment by bijeebuss2** [comment]
ok, if eval works reliably then im fine just using that

**Comment by bijeebuss2** [comment]
actually for now Im using jq to do it like this
```
secrets_json="$(dotenvx get --format json -f ".env")"
for x in $(echo "$secrets_json" | jq -r 'keys[]'); do
    [ -z "$x" ] && continue
    VALUE=$(echo "$secrets_json" | jq -r ".$x")
    export "$x"="$VALUE"
done
```

**Comment by motdotla** [comment]
yeah probably a good move to use jq since json is going to be serialized correctly

can I ask, why not use `dotenvx run` though as a simple wrapper on your processes? what's your context or preference for using `export` ?

**Comment by imme-emosol** [comment]
> ```
> $ env $(dotenvx get --format=shell)
> ...
> A=one two three
> ```
> 
> ^ this correctly prints `one two three`

Does it? Because in my zsh it seems not to, but perhaps that is because of what i do.

Given:

```sh
test ! -f .env && <<'EOF' >.env
T=some with space without quotes in dotenv
T2=" and more with spaces with double quotes in dotenv"
T3=' and w/ single quotes in dotenv'
T4=" and \" escaped double quote in double quote"
T5= and \" escaped double without quotes
T6= and \' escaped single without quotes
TEST2=4
AW=H
EOF
```

When: `\env -i $( dotenvx get --format shell ) \env`
Then: `env: ‘with’: No such file or directory`

When: `\env -i "$( dotenvx get --format shell )" \env`
Then: `T=some with space without quotes in dotenv T2= and more with spaces with double quotes in dotenv T3= and w/ single quotis in dotenv T4= and \" escaped double quote in double quote T5=and \" escaped double without quotes T6=and \' escaped single without quotes TEST2=4 AW=H`

---

Setting IFS to $'\n' (or $'\0' ), and using that in dotenvx' output (or perhaps support something like `--csv=$'\0'` so the (shell) word separator character (IFS) can be provided by the dotenvx user.

```sh
(
  export IFS=$'\n'
  \env -i A=4 $(
    export IFS=$'\n'
    for i in $( cat .env )
    do
      printf '%s=%s\n' "${i%%=*}" "${i#*=}"
    done
  ) Z=12 sh -c ' echo -e "\n.>T3:$T{3?}<_\n_>T5:${T5?}<.\n" ; \env ; '
)
```

**Comment by skycaptain** [comment]
> the only truly reliable solution right now is to use eval with --format=eval.
>
> eval $(dotenvx get --format=eval)

Hmm, when I try to run it like this, I get `sh: {MYVAR:a}: command not found`.

> can I ask, why not use dotenvx run though as a simple wrapper on your processes? what's your context or preference for using export ?

I need to merge and prepare different .env files for use in a minimal container that doesn't have tools like dotenvx or jq installed. Specifically, I want to run the dotenvx command as container during a CI pipeline, output all environment variables as a string, save this as a job artifact, and then use this string in a separate job that runs a minimal container image without dotenv or jq.

**Comment by motdotla** [comment]
>  I want to run the dotenvx command as container during a CI pipeline

you can run dotenvx in a container. there's an image here: https://dotenvx.com/docs/install#docker

**Comment by skycaptain** [comment]
> you can run dotenvx in a container. there's an image here: https://dotenvx.com/docs/install#docker

I'm already using the `dotenv/dotenvx` image in my CI pipeline. However, I need to "render" the environment variables from a `dotenv/dotenvx` container into a format that a different container—one with only a shell installed—can use. For example (rough, non-working code to illustrate):

```yaml
jobA:
  image: dotenv/dotenvx:latest
  script: dotenvx get --format=shell > shell-envs
  artifact: shell-envs

jobB:
   dependsOn: jobA
   image: busybox:stable
   script: env $(cat shell-envs) ...
```
